docs/docsite/rst/guide_test_helper.rst

```diff
@@ -0,0 +1,385 @@
+..
+  Copyright (c) Ansible Project
+  GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+  SPDX-License-Identifier: GPL-3.0-or-later
+
+.. _ansible_collections.community.general.docsite.guide_test_helper:
+
+Test Helper Guide
+=================
+
+Introduction
+^^^^^^^^^^^^
+
+The test helper was written to reduce the boilerplate code used in unit tests for modules.
+It was originally written to handle tests of modules that run external commands using ``AnsibleModule.run_command()``.
+At the time of writing (Feb 2025) that remains the only type of tests you can use
+test Helper for, but it aims to provide support for other types of interactions.
```

Comment:
The capitalization `test Helper` is somewhat strange. SHould it be "the test helper", "the Test Helper", "Test Helper", or "test helper"?

Reply:
Good question, I have asked it myself a couple of times.

Firstly, neither the file nor the class can begin with `test` or `Test`, respectively, otherwise `pytest` will try and pick them up as _regular_ test components. That being said, `helper` (in any capitalization) is quite a generic and ambiguous term.

Since it is a helper _for unit tests_, I am thinking `uthelper.py` and `UTHelper`. What do you think?

Reply:
Would it make sense, in the future - after this is more stabilised - to move it to module_utils so it can be reused in different collections?

Reply:
Sorry, forgot to reply earlier... Ah, `Helper` means the code class `Helper`, I somewhat missed that (ETOOOBVIOUS). Then maybe you should put it in double backticks so it's formatted accordingly :)

About moving it to module_utils: I don't think that makes sense, since it's about tests, not something that modules (or other plugins) should use directly. Also we can declare part of `tests/` as public API so that other collections can use it. For example basically all of community.internal_test_tools's unit test files are public API, since it contains helpers for other collections.

Reply:
I merged the other commit renaming it to `uthelper.UTHelper` and updated all the references here in the guide.